### PR TITLE
Example added for paginating results while querying using .

### PR DESCRIPTION
**Pull request by pratuat**
## Description
Please describe the change you have made.

## Checklist:
- [ ] Tests added/updated.
- [x] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [ ] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [ ] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite/client/_api/data_modeling/graphql.py [hunk 1]
```diff
@@ -156,6 +156,27 @@ def query(self, id: DataModelIdentifier, query: str, variables: dict[str, Any] |
...
+                ...     listThings (
+                ...         first: 10
+                ...         after: '<previous endCursor>'
+                ...     ) {
+                ...         items {
+                ...             thingProperty
+                ...         }
+                ...         pageInfo {
+                ...             endCursor
+                ...             hasNextPage
```
Comment (erlendvollset) [inline cognite/client/_api/data_modeling/graphql.py:172]: This `hasNextPage` is deprecated, should only rely on the existence of an `endCursor`!

Comment (erlendvollset) [reply cognite/client/_api/data_modeling/graphql.py:172]: https://github.com/cognitedata/doc.cognitedata.com/pull/1914

File: cognite/client/_api/data_modeling/graphql.py
Comment (haakonvt) [inline cognite/client/_api/data_modeling/graphql.py:1]: Nice @pratuat ! Would be even better if you show the code that does the iteration as well 🤩

Comment (haakonvt) [reply cognite/client/_api/data_modeling/graphql.py:1]: @pratuat ping 😄

**Comment by haakonvt** [comment]
@pratuat closing now, feel free to open again if you have time to finish this

**Pull request closed without merging**